framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -417,19 +418,61 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
             }
         }
 
-        // Check type-system specific annotations.
-        Class<PostconditionAnnotation> metaAnnotation = PostconditionAnnotation.class;
+        // Check for type-system specific annotations.
         List<Pair<AnnotationMirror, AnnotationMirror>> declAnnotations =
-                factory.getDeclAnnotationWithMetaAnnotation(methodElement, metaAnnotation);
+                factory.getDeclAnnotationWithMetaAnnotation(
+                        methodElement, PostconditionAnnotation.class);
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
+
+            result.addAll(getPostConditionAnnotations(anno, metaAnno));
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns a collection of postcondition annotations that can consist of
+     * {@code @EnsuresNonNull},{@code @EnsuresKeyFor},{@code @EnsuresLTLengthOf},
+     * {@code @EnsuresLockHeld} or any other postcondition annotation developed in the future.
+     *
+     * @param anno wrapper annotation if multiple annotations are present, otherwise single
+     *     annotation
+     * @param metaAnno postcondition meta-annotation that indicates that an annotation is a
+     *     postcondition annotation
+     */
+    private Set<Postcondition> getPostConditionAnnotations(
+            AnnotationMirror anno, AnnotationMirror metaAnno) {
+
+        @SuppressWarnings("unchecked")
+        List<AnnotationValue> elementvalue =
```

Comment:
This name is not very helpful, because it's just the same as the method that returned it.  Since this is the `value` field of the variable `anno`, `annoValue` would be more appropriate.

Reply:
Changed `elementvalue` to `annovalue`.